src/main/java/com/tahera/service/BookService.java

```diff
@@ -39,6 +40,17 @@ public void update(Book book) {
             bookRepository.save(updatedBook);
         }
     }
+
+    public void setBookRead(Book book){
+        Optional<Book> optionalBook = bookRepository.findByTitleAndAuthor(book.getTitle(), book.getAuthor());
+        if (optionalBook.isPresent()) {
+            Book setBookRead = optionalBook.get();
+            setBookRead.setRate(book.getRate());
+            setBookRead.setRead(true);
+            bookRepository.save(setBookRead);
+        }
```

Comment:
Since you don't consider book not being available, may i suggest making this method return a boolean instead, not being present can return false and you can check that condition on the controller to return corresponding Response with it.